internal/exec/stages/disks/partitions.go

```diff
@@ -47,25 +55,270 @@ func (s stage) createPartitions(config types.Config) error {
+// matching the partition numbers specified, but with the start and size information as determined by sgdisk.
+// The partition numbers need to passed in because sgdisk include them in it's output.
+func parseSgdiskPretend(sgdiskOut string, partitionNumbers []int) (map[int]sgdiskOutput, error) {
+	if len(partitionNumbers) == 0 {
+		return nil, nil
+	}
+	startRegex := regexp.MustCompile("^First sector: (\\d*) \\(.*\\)$")
+	endRegex := regexp.MustCompile("^Last sector: (\\d*) \\(.*\\)$")
+	const (
+		START             = iota
+		END               = iota
+		FAIL_ON_START_END = iota
+	)
+
+	output := map[int]sgdiskOutput{}
+	state := START
+	current := sgdiskOutput{}
+	i := 0
 
-			for _, part := range dev.Partitions {
-				op.CreatePartition(part)
+	lines := strings.Split(sgdiskOut, "\n")
+	for _, line := range lines {
+		switch state {
+		case START:
+			start, err := parseLine(startRegex, line)
+			if err != nil {
+				return nil, err
+			}
+			if start != -1 {
+				current.start = start
+				state = END
+			}
+		case END:
+			end, err := parseLine(endRegex, line)
+			if err != nil {
+				return nil, err
+			}
+			if end != -1 {
+				current.size = 1 + end - current.start
+				output[partitionNumbers[i]] = current
+				i++
+				if i == len(partitionNumbers) {
+					state = FAIL_ON_START_END
+				} else {
+					current = sgdiskOutput{}
+					state = START
+				}
 			}
+		case FAIL_ON_START_END:
+			if len(startRegex.FindStringSubmatch(line)) != 0 ||
+				len(endRegex.FindStringSubmatch(line)) != 0 {
+				return nil, ErrBadSgdiskOutput
+			}
+		}
+	}
+
+	if state != FAIL_ON_START_END {
+		// We stopped parsing in the middle of a info block. Something is wrong
+		return nil, ErrBadSgdiskOutput
+	}
+
+	return output, nil
+}
+
+// partitionShouldExist returns whether a bool is indicating if a partition should exist or not.
+// nil (unspecified in json) is treated the same as true.
+func partitionShouldExist(part types.Partition) bool {
+	return part.ShouldExist == nil || *part.ShouldExist
+}
 
-			if err := op.Commit(); err != nil {
-				return fmt.Errorf("commit failure: %v", err)
+// getPartitionMap returns a map of partitions on device, indexed by partition number
+func (s stage) getPartitionMap(device string) (map[int]types.Partition, error) {
+	parts := []types.Partition{}
+	err := s.Logger.LogOp(
+		func() error {
+			p, err := util.DumpPartitionTable(device)
+			if err != nil {
+				return err
 			}
+			parts = p
 			return nil
-		}, "partitioning %q", devAlias)
-		if err != nil {
-			return err
+		}, "reading partition table of %q", device)
+	if err != nil {
+		return nil, err
+	}
+	m := map[int]types.Partition{}
+	for _, part := range parts {
+		m[part.Number] = part
+	}
+	return m, nil
+}
+
+// partitionDisk partitions devAlias according to the spec given by dev
+func (s stage) partitionDisk(dev types.Disk, devAlias string) error {
+	if dev.WipeTable {
```

Comment:
What's the reason to not wipe during the following operation?

Reply:
`getPartitionMap` reads the disk and needs to see that it's empty